cmd/telemetrygen/internal/traces/config.go

```diff
@@ -36,4 +50,15 @@ func (c *Config) Flags(fs *pflag.FlagSet) {
 	fs.BoolVar(&c.Batch, "batch", true, "Whether to batch traces")
 	fs.IntVar(&c.LoadSize, "size", 0, "Desired minimum size in MB of string data for each trace generated. This can be used to test traces with large payloads, i.e. when testing the OTLP receiver endpoint max receive size.")
 	fs.DurationVar(&c.SpanDuration, "span-duration", 123*time.Microsecond, "The duration of each generated span.")
+
+	// TLS CA configuration
+	fs.StringVar(&c.CaFile, "ca-cert", "", "Trusted Certificate Authority to verify collector receiver certificate")
+
+	// mTLS configuration
+	fs.BoolVar(&c.ClientAuth.Enabled, "mtls", false, "Whether to require client authentication for mTLS")
+	fs.StringVar(&c.ClientAuth.ClientCertFile, "client-cert", "", "Client certificate file")
+	fs.StringVar(&c.ClientAuth.ClientKeyFile, "client-key", "", "Client private key file")
```

Comment:
The current implementation only supports these for traces. Can we make these supported on metrics and logs as well? For metrics, it should be very similar to traces, just editing `cmd/telemetrygen/internal/metrics/exporter.go`. Logs is a bit different, but probably not hard either, needs updates on https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/4255836c67b8bd5528a3952de954c9ca3612b335/cmd/telemetrygen/internal/logs/exporter.go

Reply:
Thanks @mx-psi  for your feedback. It's possible to enable TLS/mTLS for metrics and logs as well. I'm working on these right now.